src/app/ngrx/reducers/index.ts

```diff
@@ -24,18 +26,23 @@ export const reducers: ActionReducerMap<RootState> = {
 
 export const selectAppState = (state: RootState) => state;
 
-export const selectFilter = createFeatureSelector<FilterModel>('filter');
-
-// these selectors off the selectAppState selector should only emit changes when that slice of the filter state changes, handy!
-export const selectPodcastFilter = createSelector(selectAppState, (state: RootState) => state.filter ? state.filter.podcastSeriesId : undefined);
-
-export const selectEpisodeFilter = createSelector(selectAppState, (state: RootState) => state.filter ? state.filter.episodeIds : undefined);
-
-export const selectIntervalFilter = createSelector(selectAppState, (state: RootState) => state.filter ? state.filter.interval : undefined);
-
-export const selectPodcasts = createFeatureSelector<PodcastModel[]>('podcasts');
-
-export const selectEpisodes = createFeatureSelector<EpisodeModel[]>('episodes');
+export const selectFilter = createSelector(selectAppState, (state: RootState) => state.filter);
+export const selectPodcastFilter = createSelector(selectFilter, (filter: FilterModel) => filter.podcastSeriesId);
+export const selectPageFilter = createSelector(selectFilter, (filter: FilterModel) => filter.page);
+export const selectIntervalFilter = createSelector(selectFilter, (filter: FilterModel) => filter.interval);
+
+export const selectPodcastState = createSelector(selectAppState, (state: RootState) => state.podcasts);
+export const selectPodcastEntities = createSelector(selectPodcastState, getPodcastEntities);
+export const selectPodcasts = createSelector(selectPodcastEntities, entities => {
+  return Object.keys(entities).map(seriesId => entities[parseInt(seriesId, 10)]);
+});
+export const selectPodcastsError = createSelector(selectPodcastState, getPodcastError);
+
+export const selectEpisodeState = createSelector(selectAppState, (state: RootState) => state.episodes);
+export const selectEpisodeEntities = createSelector(selectEpisodeState, getEpisodeEntities);
```

Comment:
does this (and ditto for Podcast equiv) need to be exported if it's only used by another fn in here? Do you anticipate other places calling it?

Reply:
I don't expect entities to be used outside of other selectors, but it's a pattern I'm working on following that these are exported. This entity pattern will allow me to turn the filter util functions into actual selectors; just the thing I was looking for initially to get smaller slices of state into selectors. With this entity pattern for example, we could have a selector that gives us a sorted list of episodes and remove that logic from components. I _think_ we should also be able to map data to the timeseries models that the chart uses in those selectors. It's a work in progress. and I'm still not quite sure how I'm going to fit the work in based on priority tbh, but that's the thought behind it.